client/my-sites/marketplace/pages/marketplace-plugin-upload-status/index.tsx

```diff
@@ -0,0 +1,107 @@
+import getPluginUploadError from 'calypso/state/selectors/get-plugin-upload-error';
+import getPluginUploadProgress from 'calypso/state/selectors/get-plugin-upload-progress';
+import getUploadedPluginId from 'calypso/state/selectors/get-uploaded-plugin-id';
+import isPluginActive from 'calypso/state/selectors/is-plugin-active';
+import isPluginUploadComplete from 'calypso/state/selectors/is-plugin-upload-complete';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import './style.scss';
+
+const MarketplacePluginUpload = (): JSX.Element => {
+	const [ currentStep, setCurrentStep ] = useState( 0 );
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
+	const siteId = useSelector( getSelectedSiteId ) as number;
+	const pluginUploadProgress = useSelector( ( state ) => getPluginUploadProgress( state, siteId ) );
+	const pluginUploadError = useSelector( ( state ) => getPluginUploadError( state, siteId ) );
+	const uploadedPluginSlug = useSelector( ( state ) =>
+		getUploadedPluginId( state, siteId )
+	) as string;
+	const pluginUploadComplete = useSelector( ( state ) => isPluginUploadComplete( state, siteId ) );
+	const uploadedPlugin = useSelector( ( state ) =>
+		getPluginOnSite( state, siteId, uploadedPluginSlug )
+	);
+	const pluginActive = useSelector( ( state ) =>
+		isPluginActive( state, siteId, uploadedPluginSlug )
+	);
+	const automatedTransferStatus = useSelector( ( state ) =>
+		getAutomatedTransferStatus( state, siteId )
+	);
+
+	useEffect( () => {
+		if ( 100 === pluginUploadProgress ) {
+			// For smaller uploads or fast networks give
+			// the chance to Upload Plugin step to be shown
+			// before moving to next step.
+			waitFor( 1 ).then( () => setCurrentStep( 1 ) );
+		}
+	}, [ pluginUploadProgress, setCurrentStep ] );
+
+	useEffect( () => {
+		if ( pluginUploadComplete && uploadedPlugin ) {
+			dispatch(
+				togglePluginActivation( siteId, {
+					slug: uploadedPlugin?.slug,
+					id: uploadedPlugin?.id,
+				} )
+			);
+			setCurrentStep( 2 );
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ pluginUploadComplete, uploadedPlugin, setCurrentStep ] );
+
+	useEffect( () => {
+		if ( pluginActive ) {
+			waitFor( 1 ).then( () =>
+				page( `/marketplace/thank-you/${ uploadedPlugin?.slug }/${ selectedSiteSlug }` )
+			);
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ pluginActive, automatedTransferStatus ] ); // We need to trigger this hook also when automatedTransferStatus changes.
+
+	const steps = [
+		translate( 'Uploading plugin' ),
+		translate( 'Installing plugin' ),
+		translate( 'Activating plugin' ),
+	];
+
+	return (
+		<ThemeProvider theme={ theme }>
+			<PageViewTracker path="/marketplace/product/install/:site" title="Plugins > Installing" />
+			{ siteId ? <QueryJetpackPlugins siteIds={ [ siteId ] } /> : '' }
```

Comment:
Nitpick (non-blocking): I think this reads better:
```suggestion
			{ siteId && <QueryJetpackPlugins siteIds={ [ siteId ] } /> }
```